runtime/gc_api/HeapIteratorAPI.cpp

```diff
@@ -546,6 +547,17 @@ j9mm_iterate_all_continuation_objects(J9VMThread *vmThread, J9PortLibrary *portL
 	J9MM_IterateRegionDescriptor regionDesc;
 	jvmtiIterationControl returnCode = JVMTI_ITERATION_CONTINUE;
 
+	MM_EnvironmentBase *env = MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread);
+	UDATA cycle_type = env->_cycleState->_type;
+
+	if ((OMR_GC_CYCLE_TYPE_SCAVENGE == cycle_type) || (OMR_GC_CYCLE_TYPE_VLHGC_PARTIAL_GARBAGE_COLLECT == cycle_type) ||
+		(OMR_GC_CYCLE_TYPE_GLOBAL   == cycle_type) || (OMR_GC_CYCLE_TYPE_VLHGC_GLOBAL_GARBAGE_COLLECT  == cycle_type)) {
+		if (extensions->didIterateContinuationListForJIT) {
+			extensions->didIterateContinuationListForJIT = false;
```

Comment:
I hope we can get by without introducing this flag by registering JIT reclamation pass to CYCLE_END for all type of collectors (not just realtime). By doing that, we guarantee that stack walking pass for JIT (that we do now in clearable phase) is always done before reclamation pass. Clearable pass is done in the last increment of a cycle (if a cycle has more than one increment like concurrent scavenge, GMP, or realtime) when cycle end hook is also triggered that could do the reclamation.